apps/rush-lib/src/api/RushConfiguration.ts

```diff
@@ -352,6 +360,22 @@ export class PnpmOptionsConfiguration extends PackageManagerOptionsConfiguration
    */
   public readonly preventManualShrinkwrapChanges: boolean;
 
+  /**
+   * If true, then Rush will use the workspaces feature to install and link packages when invoking PNPM.
+   *
+   * @remarks
+   * The default value is false.  (For now.)
+   */
+  public readonly useWorkspaces: boolean;
+
+  /**
+   * If true, then Rush will shim common versions support into the pnpmfile.
```

Comment:
What does this mean? We should document it more clearly

Reply:
Pnpm does not apply common versions to workspace projects in the same way as it used to, since it's not all performed as a single install. The PNPM shim file is specified in the BaseInstallManager. Basically, what it does is create a pnpmfile.js in common/temp. This file contains all the calculated common versions, as well as the allowedAlternateVersions. For each package, we go through the dependencies (reg, dev, opt) and check to see if we have a common version specified for that dependency. If we do, and it's not an allowedAlternateVersion, AND the common version range is a subset of the specified range, we substitute the version specifier for the common version range. We then call into the client-provided pnpmfile that is checked into `common/config/rush` to run the normal pnpmfile functionality. This way, we can shim in common-versions support to all packages that we consume, and it makes common versions much more stable, especially in the case of workspaces (though there is no reason that it needs to be limited to workspaces)

Reply:
@D4N14L when you say "common versions" do you mean `preferredVersions`?

Reply:
@octogonz Correct. We also attempt to use the implicit preferred versions, so not just the `preferredVersions` specified in `common-versions.json`